tests/tensorizer_loader/test_tensorizer.py

```diff
@@ -357,29 +383,30 @@ def test_assert_deserialization_kwargs_passed_to_tensor_deserializer(tmp_path, c
     args = EngineArgs(model=model_ref, device="cuda")
     tensorize_vllm_model(args, config)
 
-    loader_tc = TensorizerConfig(
-        tensorizer_uri=str(model_path),
-        deserialization_kwargs=deserialization_params,
-    )
-
     # lazy_load = True is listed as a deserialization_param, which will
     # break the loading process because is_vllm_tensorized hard-codes a
     # lazy_load = True to check for the vllm-tensorized marker.
 
     # If `deserialization_params` is passed to TensorDeserializer as expected,
     # we should expect the ValueError being caught.
-    try:
-        LLM(
-            model=model_ref,
-            load_format="tensorizer",
-            model_loader_extra_config=loader_tc,
-        )
-    except RuntimeError:
-        out, err = capfd.readouterr()
-        combined_output = out + err
-        assert ("TypeError: tensorizer.serialization.TensorDeserializer() got multiple values for keyword argument 'lazy_load'") in combined_output
 
 
+    loader_tc = TensorizerConfig(
+        tensorizer_uri=str(model_path),
+        deserialization_kwargs=deserialization_params,
+    )
+
+    engine_args = EngineArgs(
+        model="facebook/opt-125m",
+        load_format = "tensorizer",
+        model_loader_extra_config=loader_tc.to_dict(),)
+
+    vllm_config = engine_args.create_engine_config()
+    executor = DummyExecutor(vllm_config)
+
+    with pytest.raises(TensorizerCaughtError):
+        executor.collective_rpc(patch_init_and_catch_error, args=(tensorizer.serialization.TensorDeserializer, "__init__",  TypeError,))
```

Comment:
You should leave a comment that you're translating `TypeError`s into `TensorizerCaughtError` first instead of just checking `pytest.raises(TypeError)` directly to narrow the scope so that it only passes this test when `TensorDeserializer.__init__()` is the method throwing a `TypeError`.
At least, that's my reading of why you're doing that.